core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -63,11 +68,11 @@
   private final SerializingExecutor callExecutor;
   private final boolean unaryRequest;
   private final CallOptions callOptions;
-  private ClientStream stream;
+  private DelayedStream stream;
```

Comment:
Always using `DelayedStream` keep the failure logic inside `DelayedStream`, which I like better. On the fast path, the overhead of `DelayedStream` is just a function call and a volatile read, which should be negligible.